docs/ai/reference_extai.rst

```diff
@@ -0,0 +1,403 @@
+
+    * - :eql:func:`ext::ai::search`
+      - :eql:func-desc:`ext::ai::search`
+
+
+------------
+
+
+.. eql:function:: ext::ai::to_context(object: anyobject) -> str
+
+    Returns the indexed expression value for an object with an ``ext::ai::index``.
+
+    **Example**:
+
+    Schema:
+
+    .. code-block:: sdl
+
+        module default {
+          type Astronomy {
+            topic: str;
+            content: str;
+            deferred index ext::ai::index(embedding_model := 'text-embedding-3-small')
+              on (.topic ++ ' ' ++ .content);
+          }
+        };
+
+    Data:
+
+    .. code-block:: edgeql-repl
+
+        db> insert Astronomy {
+        ...   topic := 'Mars',
+        ...   content := 'Skies on Mars are red.'
+        ... }
+        db> insert Astronomy {
+        ...   topic := 'Earth',
+        ...   content := 'Skies on Earth are blue.'
+        ... }
+
+    Results of calling ``to_context``:
+
+    .. code-block:: edgeql-repl
+
+        db> select ext::ai::to_context(Astronomy);
+
+        {'Mars Skies on Mars are red.', 'Earth Skies on Earth are blue.'}
+
+
+------------
+
+
+.. eql:function:: ext::ai::search( \
+                    object: anyobject, \
+                    query: array<float32> \
+                  ) -> optional tuple<object: anyobject, distance: float64>
+
+    Searches objects using their :ref:`ai::index
+    <ref_ai_extai_reference_index>`.
+
+    Returns tuples of (object, distance).
+
+    .. note::
+
+        The ``query`` argument should *not* be a textual query but the
+        embeddings generated *from* a textual query.
+
+    .. code-block:: edgeql-repl
+
+        db> with query := <array<float32>><json>$query
+        ... select ext::ai::search(Knowledge, query);
+
+        {
+          (
+            object := default::Knowledge {id: 9af0d0e8-0880-11ef-9b6b-4335855251c4},
+            distance := 0.20410746335983276
+          ),
+          (
+            object := default::Knowledge {id: eeacf638-07f6-11ef-b9e9-57078acfce39},
+            distance := 0.7843298847773637
+          ),
+          (
+            object := default::Knowledge {id: f70863c6-07f6-11ef-b9e9-3708318e69ee},
+            distance := 0.8560434728860855
+          ),
+        }
+
+
+Types
+=====
+
+Provider Configuration Types
+----------------------------
+
+Provider configurations are required for AI indexes and RAG functionality.
+
+Example provider configuration:
+
+.. code-block:: edgeql
+
+    configure current database
+    insert ext::ai::OpenAIProviderConfig {
+      secret := 'sk-....',
+    };
+
+.. note::
+
+    All provider types require the ``secret`` property be set with a string
+    containing the secret provided by the AI vendor.
+
+
+.. note::
+
+    ``ext::ai::CustomProviderConfig requires an ``api_style`` property be set.
+
+ext::ai::ProviderAPIStyle
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Enum defining supported API styles:
+
+* ``OpenAI``
+* ``Anthropic``
+
+ext::ai::ProviderConfig
```

Comment:
This can be in a follow-up PR, but we should show what custom providers look like somewhere.